src/pyhf/workspace.py

```diff
@@ -161,3 +163,195 @@ def data(self, model, with_aux=True):
+        """
+        Return a new, pruned workspace specification. This will not modify the original workspace.
+
+        The pruned workspace must also be a valid workspace.
+
+        Args:
+            modifiers: A :obj:`str` or a :obj:`list` of modifiers to prune.
+            modifier_types: A :obj:`str` or a :obj:`list` of modifier types to prune.
+            samples: A :obj:`str` or a :obj:`list` of samples to prune.
+            channels: A :obj:`str` or a :obj:`list` of channels to prune.
+            measurements: A :obj:`str` or a :obj:`list` of measurements to prune.
+
+        Returns:
+            ~pyhf.workspace.Workspace: A new workspace object with the specified components removed
+
+        """
+        return self._prune_and_rename(
+            prune_modifiers=modifiers,
+            prune_modifier_types=modifier_types,
+            prune_samples=samples,
+            prune_channels=channels,
+            prune_measurements=measurements,
+        )
+
+    def rename(self, modifiers={}, samples={}, channels={}, measurements={}):
+        """
+        Return a new workspace specification with certain elements renamed. This will not modify the original workspace.
+
+        The renamed workspace must also be a valid workspace.
+
+        Args:
+            modifiers: A :obj:`dict` mapping old modifier name to new modifier name.
+            samples: A :obj:`dict` mapping old sample name to new sample name.
+            channels: A :obj:`dict` mapping old channel name to new channel name.
+            measurements: A :obj:`dict` mapping old measurement name to new measurement name.
+
+        Returns:
+            ~pyhf.workspace.Workspace: A new workspace object with the specified components renamed
+
+        """
+        return self._prune_and_rename(
+            rename_modifiers=modifiers,
+            rename_samples=samples,
+            rename_channels=channels,
+            rename_measurements=measurements,
+        )
+
+    @classmethod
+    def combine(cls, left, right):
+        """
+        Return a new workspace specification that is the combination of the two workspaces.
+
+        The new workspace must also be a valid workspace.
+
+        A combination of workspaces is done by joining the set of channels. If the workspaces share any channels or measurements in common, do not combine.
+
+        Args:
+            left (~pyhf.workspace.Workspace): A workspace
+            right (~pyhf.workspace.Workspace): Another workspace
+
+        Returns:
+            ~pyhf.workspace.Workspace: A new combined workspace object
+
+        """
+        common_channels = set(left.channels).intersection(right.channels)
+        if common_channels:
+            raise exceptions.InvalidWorkspaceOperation(
+                "Workspaces cannot have any channels in common: {}".format(
+                    common_channels
+                )
+            )
+
+        common_measurements = set(left.measurement_names).intersection(
```

Comment:
the common POI is fine, but we didn't discuss how to handle merging parameter configurations under a given measurement.. and that's a bit complicated because combining workspaces might end up with configuring a given parameter two different ways...

Reply:
don't we have the paramset reduction for that to check compatibility?

Reply:
Good point. I've removed common measurements and just checked for incompatible POI.